tests/test-webapps/test-websocket-client-webapp/src/main/java/org/eclipse/jetty/tests/webapp/websocket/WebSocketClientServlet.java

```diff
@@ -0,0 +1,125 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.tests.webapp.websocket;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketClose;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketConnect;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketMessage;
+import org.eclipse.jetty.websocket.api.annotations.WebSocket;
+import org.eclipse.jetty.websocket.api.util.WSURI;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+
+@WebServlet("/")
+public class WebSocketClientServlet extends HttpServlet
+{
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException
+    {
+
+        WebSocketClient client = null;
+        try
+        {
+            client = new WebSocketClient();
+            WebSocketClient finalClient = client;
+            runThrowExceptionsAsRuntime(() -> finalClient.start());
+            resp.setContentType("text/html");
+            //resp.getWriter().println("ConnectTimeout: " + client.getHttpClient().getConnectTimeout());
```

Comment:
With these changes the WebSocketClient can load and use the jetty-websocket-httpclient XML file with server class access, but we still don't have access to classes like `HttpClient` in the webapp. So this line will result in an exception.

And if instead of doing the line `runThrowExceptionsAsRuntime(() -> finalClient.start());` and I do as intellij suggests and use `runThrowExceptionsAsRuntime(finalClient::start);` then I get an exception as it doesn't know `LifeCycle`.

Reply:
I don't think you need `ThrowingRunnable` (can't you use `Callable` instead?).
Plus it's not clear what line is the problem -- there should be a comment in the code if it's non trivial due to class loading.
I'd like an explanation that there are 2 cases (webapp classes vs server classes), and what happens in each case.

Reply:
`Callable` doesn't work because it still throws exception. I guess I could just put a nested try catch block that wraps the exception with `RuntimeException`. I just thought it would be cleaner with the method reference and was surprised that it didn't work.